### PR TITLE
Change OD unity test

### PR DESCRIPTION
**Pull request by jakesabathia2**
close https://github.com/apple/turicreate/issues/2441.

### CODE REVIEW

**Comment by nickjong** [comment]
@Jarvi-Izana Here's an example of a nice unit test for (in this case) the OD model definition

File: test/unity/toolkits/object_detection/test_od_serialization.cxx [hunk 1]
```diff
@@ -30,7 +30,7 @@ BOOST_AUTO_TEST_CASE(test_init_darknet_yolo) {
   init_darknet_yolo(nn_spec, num_classes, anchor_boxes);
 
   const CoreML::Specification::NeuralNetwork& nn = nn_spec.get_coreml_spec();
-  TS_ASSERT_EQUALS(nn.layers_size(), 25);
+  TS_ASSERT_EQUALS(nn.layers_size(), 31);
```
Comment (shantanuchhabra) [inline test/unity/toolkits/object_detection/test_od_serialization.cxx]: Could you add a comment explaining the magic number here?

File: test/unity/toolkits/object_detection/test_od_serialization.cxx [hunk 2]
```diff
@@ -62,7 +62,29 @@ BOOST_AUTO_TEST_CASE(test_init_darknet_yolo) {
                      "leakyrelu" + std::to_string(x.first) + "_fwd");
     TS_ASSERT_EQUALS(relulayer_.activation().leakyrelu().alpha(), 0.1f);
 
-    layer_num = layer_num + 3;
+    if (x.first <= 5){
+      const auto& poolinglayer_ = nn.layers(layer_num + 3);
```
Comment (shantanuchhabra) [inline test/unity/toolkits/object_detection/test_od_serialization.cxx]: There seem to be a lot of magic numbers in this file. Could you add comments (or make variables) explaining what the magic numbers mean, so the code becomes more readable and inheritable?

**Comment by jakesabathia2** [comment]
Just add the test for the final conversion layer which wasn't covered in the past.
And also make some comments :)

**Comment by Jarvi-Izana** [comment]
> @Jarvi-Izana Here's an example of a nice unit test for (in this case) the OD model definition

Will have a look.

**Pull request merged**